hazelcast/src/main/java/com/hazelcast/collection/impl/queue/operations/CheckAndEvictOperation.java

```diff
@@ -37,8 +40,10 @@ public CheckAndEvictOperation(String name) {
     public void run() throws Exception {
         final QueueContainer queueContainer = getContainer();
         if (queueContainer.isEvictable()) {
-            ProxyService proxyService = getNodeEngine().getProxyService();
-            proxyService.destroyDistributedObject(getServiceName(), name);
+            NodeEngine nodeEngine = getNodeEngine();
+            ProxyService proxyService = nodeEngine.getProxyService();
+            UUID source = nodeEngine.getLocalMember().getUuid();
```

Comment:
We give running node's uuid here, shouldn't it be caller's uuid?

Reply:
Yes, but for this operation the caller uuid is not being filled. It can be done both ways, either set caller uuid where this op is created and get the caller uuid at this point and use it or just use the change I did which seemed simpler to me.

Reply:
As far as i see, this operation's running target is always local partition which holds queue. So it seems what you are doing is the correct way to set uuid.